eureka-core/src/main/java/com/netflix/eureka/registry/ResponseCache.java

```diff
@@ -0,0 +1,42 @@
+package com.netflix.eureka.registry;
+
+import javax.annotation.Nullable;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * @author David Liu
+ */
+public interface ResponseCache {
+
+    public enum KeyType {
+        JSON, XML
+    }
+
+    void invalidate(String appName, @Nullable String vipAddress, @Nullable String secureVipAddress);
+
+    AtomicLong getVersionDelta();
+
+    AtomicLong getVersionDeltaWithRegions();
+
+    /**
+     * Get the cached information about applications.
+     *
+     * <p>
+     * If the cached information is not available it is generated on the first
+     * request. After the first request, the information is then updated
+     * periodically by a background thread.
+     * </p>
+     *
+     * @param key the key for which the cached information needs to be obtained.
+     * @return payload which contains information about the applications.
+     */
+    <KEY> String get(KEY key);
```

Comment:
If it at the class level then it'll need to be propagated all the way up to the PeerAwareRegistry, which gets messy. I'm not happy with the way either. Maybe this is over wrapping it and we should just let class Key be part of the interface.